spec/fixtures/external_components/components/logger.rb

```diff
@@ -7,11 +7,26 @@ class Logger
         class << self
           attr_accessor :default_level
         end
+
+        self.default_level = :scream
+
+        attr_reader :log_level
+
+        def initialize(log_level = Logger.default_level)
+          @log_level = log_level
+        end
       end
     end
   end
 
   start do
-    register(:logger, ExternalComponents::Logger.new)
+    logger =
+      if config
+        ExternalComponents::Logger.new(config.log_level)
+      else
+        ExternalComponents::Logger.new
+      end
```

Comment:
I'm wondering if we should try and move to a slightly more formal configuration structure, either here in this PR or in another before we release this.

I feel like it would be good to have declared configuration keys along with defaults. It'd let external bootable component authors expose a lot more things in config and avoid having these if/else things throughout the component's own lifecycle code.

e.g. if we declared something like `setting :log_level, :info`, the logger could be set up like `ExternalComponents::Logger.new(config.log_level)` regardless of whether the application author provided a custom log level or not.

Reply:
@timriley yeah I agree, I'll improve this in a separate PR. I want to port dry-web's settings to dry-system and somehow re-use it for component configuration too